packages/sfui/frameworks/vue/composables/usePagination/usePagination.ts

```diff
@@ -0,0 +1,29 @@
+import { ref, computed } from 'vue';
+import paginate from 'jw-paginate';
+
+interface UsePaginationParameters {
+  totalItems: number;
+  currentPage?: number;
+  pageSize?: number;
+  maxPages?: number;
+}
+
+export function usePagination({ totalItems, currentPage = 1, pageSize = 10, maxPages = 1 }: UsePaginationParameters) {
+  const activePage = ref(currentPage);
+
+  const pagination = computed(() => {
+    return paginate(totalItems, activePage.value, pageSize, maxPages);
+  });
+
+  return {
+    totalPages: computed(() => pagination.value.totalPages),
+    pages: computed(() => pagination.value.pages),
+    selectedPage: computed(() => pagination.value.currentPage),
+    endPage: computed(() => pagination.value.endPage),
+    startPage: computed(() => pagination.value.startPage),
+    maxVisiblePages: maxPages,
+    next: () => (activePage.value += 1),
+    prev: () => (activePage.value -= 1),
+    onPageChange: (newPage: number) => (activePage.value = newPage),
```

Comment:
also `onPageChange` here is bit confusing. I would name function that if it was passed via prop to react to page change. here you are doing the `setPage`

Reply:
https://github.com/vuestorefront/storefront-ui/pull/2535/commits/3805a05d6d05b2db74b19ae6a9f4e6094aba1976